bake/bake.go

```diff
@@ -1274,14 +1274,15 @@ func TargetsToBuildOpt(m map[string]*Target, inp *Input) (map[string]build.Optio
 	return m2, nil
 }
 
-func updateContext(t *build.Inputs, inp *Input) {
+func updateContext(t *build.Inputs, inp *Input, target *Target) {
 	if inp == nil || inp.State == nil {
 		return
 	}
 
 	for k, v := range t.NamedContexts {
 		if v.Path == "." {
 			t.NamedContexts[k] = build.NamedContext{Path: inp.URL}
+			continue
```

Comment:
@tonistiigi I saw this was (seemingly) missing while implementing my change. Without it, the context is simply reassigned below. Let me know if my assumption isn't correct and I'll remove it.